### PR TITLE
[IMP] sale: in sale-order report taxes should be grouped by tax group.

### PR DESCRIPTION
**Pull request by kga-odoo**
<b>Task:</b> https://www.odoo.com/web#id=21850&view_type=form&model=project.task&action=327&menu_id=4720
<b>Pad:</b>https://pad.odoo.com/p/r.b7UDV2J0wKvfqXOG

### CODE REVIEW

File: addons/sale/sale.py [hunk 1]
```diff
@@ -413,6 +414,18 @@ def _create_analytic_account(self, prefix=None):
             })
             order.project_id = analytic
 
+    @api.multi
+    def _get_taxes_by_group(self):
+        self.ensure_one()
+        res = {}
+        currency = self.currency_id or self.company_id.currency_id
+        for line in self.order_line:
```
Comment (atp-odoo) [inline addons/sale/sale.py]: use self.mapped('order_line.tax_id').
for tax in self.mapped('order_line.tax_id')

Comment (kga-odoo) [inline addons/sale/sale.py]: we need line.price_subtotal at line no- 425.

File: addons/sale/sale.py [hunk 2]
```diff
@@ -413,6 +414,20 @@ def _create_analytic_account(self, prefix=None):
...
 
+    @api.multi
+    def _get_taxes_by_group(self):
+        self.ensure_one()
+        res = {}
+        for line in self.order_line:
+            for tax in line.tax_id.children_tax_ids if line.tax_id.amount_type == 'group' else line.tax_id:
+                res.setdefault(tax.tax_group_id, 0.0)
+                if tax.amount_type == 'fixed':
+                    res[tax.tax_group_id] += tax.amount
```
Comment (mdi-odoo) [inline addons/sale/sale.py]: Use line.tax_id.compute_all() method to compute the tax amount .

File: addons/sale/sale.py [hunk 3]
```diff
@@ -417,6 +418,20 @@ def _create_analytic_account(self, prefix=None):
...
 
+    @api.multi
+    def _get_taxes_by_group(self):
+        self.ensure_one()
+        Tax = self.env['account.tax']
+        res = {}
+        for line in self.order_line:
+            taxes = line.tax_id.compute_all(line.price_unit, quantity=line.product_uom_qty)['taxes']
+            for tax in taxes:
+                group = Tax.browse(tax['id']).tax_group_id
```
Comment (tivisse) [inline addons/sale/sale.py]: Rebrowsing the each tax on each line is a bad idea as it is already browsed with the line

**Pull request merged**